src/test/java/seedu/exercise/logic/commands/StatsFactoryTest.java

```diff
@@ -0,0 +1,143 @@
+package seedu.exercise.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.exercise.testutil.CommonTestData.VALID_BAR_CHART;
+import static seedu.exercise.testutil.CommonTestData.VALID_LINE_CHART;
+import static seedu.exercise.testutil.CommonTestData.VALID_PIE_CHART;
+import static seedu.exercise.testutil.CommonTestData.VALID_STATISTIC_CATEGORY_CALORIES;
+import static seedu.exercise.testutil.CommonTestData.VALID_STATISTIC_CATEGORY_EXERCISE;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.exercise.logic.commands.statistic.Statistic;
+import seedu.exercise.logic.commands.statistic.StatsFactory;
+import seedu.exercise.model.ReadOnlyResourceBook;
+import seedu.exercise.model.property.Date;
+import seedu.exercise.model.resource.Exercise;
+import seedu.exercise.testutil.builder.ExerciseBookBuilder;
+import seedu.exercise.testutil.builder.ExerciseBuilder;
+import seedu.exercise.testutil.builder.StatisticBuilder;
+
+public class StatsFactoryTest {
+
+    private static final String START_DATE = "25/09/2019";
+    private static final String END_DATE = "30/09/2019";
+    private ReadOnlyResourceBook<Exercise> eb;
+
+    @BeforeEach
+    public void setUp() {
+        Exercise running = new ExerciseBuilder().build(); //Running, 111 kcal, 0.5 hours
+        Exercise swimming = new ExerciseBuilder().withName("Swimming").withDate("27/09/2019")
+                .withCalories("300").withQuantity("10").withUnit("laps").build();
+        Exercise cycling = new ExerciseBuilder().withName("Cycling").withDate("28/09/2019")
+                .withCalories("150").withQuantity("5").withUnit("km").build();
+        Exercise anotherRunning = new ExerciseBuilder().withDate("28/09/2019").withQuantity("1.0").build();
+        eb = new ExerciseBookBuilder().withExercise(running).withExercise(swimming)
+                .withExercise(cycling).withExercise(anotherRunning).build();
+    }
```

Comment:
Not present in `CommonTestData`?

Reply:
I created new Exercise because i need different dates, but i have moved to `typicalExercise`.